src/main/java/it/gov/pagopa/cgn/portal/facade/BackofficeAttributeAuthorityFacade.java

```diff
@@ -66,10 +66,11 @@ public ResponseEntity<OrganizationWithReferentsAndStatus> getOrganization(String
     @Transactional(Transactional.TxType.REQUIRED)
     public ResponseEntity<OrganizationWithReferents> upsertOrganization(OrganizationWithReferents organizationWithReferents) {
         // find agreement for this organization and apply an update consumer
+
         agreementUserService.findCurrentAgreementUser(organizationWithReferents.getKeyOrganizationFiscalCode())
-                            .ifPresent(agreementUserEntity -> updateAgreementUserAndProfileConsumer.accept(
-                                    agreementUserEntity,
-                                    organizationWithReferents));
+                            .ifPresentOrElse(agreementUserEntity -> updateAgreementUserAndProfileConsumer.accept(agreementUserEntity, organizationWithReferents),
```

Comment:
what if on Admin side they update the entity type? Here the update is not considering it to update to the new value.

Reply:
correct, modified to update the entityType in case of an already present Agreement